test/e2e/storage/pd.go

```diff
@@ -59,355 +60,256 @@ var _ = SIGDescribe("Pod Disks", func() {
-		framework.SkipUnlessProviderIs("gce", "gke", "aws")
-
-		By("creating PD")
-		diskName, err := framework.CreatePDWithRetry()
-		framework.ExpectNoError(err, "Error creating PD")
-		numContainers := 4
-		var host0Pod *v1.Pod
-
-		defer func() {
-			By("cleaning up PD-RW test environment")
-			// Teardown pods, PD. Ignore errors.
-			// Teardown should do nothing unless test failed.
-			if host0Pod != nil {
-				podClient.Delete(host0Pod.Name, metav1.NewDeleteOptions(0))
-			}
-			detachAndDeletePDs(diskName, []types.NodeName{host0Name})
-		}()
-
-		fileAndContentToVerify := make(map[string]string)
-		for i := 0; i < 3; i++ {
-			framework.Logf("PD Read/Writer Iteration #%v", i)
-			By("submitting host0Pod to kubernetes")
-			host0Pod = testPDPod([]string{diskName}, host0Name, false /* readOnly */, numContainers)
-			_, err = podClient.Create(host0Pod)
-			framework.ExpectNoError(err, fmt.Sprintf("Failed to create host0Pod: %v", err))
-
-			framework.ExpectNoError(f.WaitForPodRunningSlow(host0Pod.Name))
-
-			// randomly select a container and read/verify pd contents from it
-			containerName := fmt.Sprintf("mycontainer%v", mathrand.Intn(numContainers)+1)
-			verifyPDContentsViaContainer(f, host0Pod.Name, containerName, fileAndContentToVerify)
-
-			// Randomly select a container to write a file to PD from
-			containerName = fmt.Sprintf("mycontainer%v", mathrand.Intn(numContainers)+1)
-			testFile := fmt.Sprintf("/testpd1/tracker%v", i)
-			testFileContents := fmt.Sprintf("%v", mathrand.Int())
-			fileAndContentToVerify[testFile] = testFileContents
-			framework.ExpectNoError(f.WriteFileViaContainer(host0Pod.Name, containerName, testFile, testFileContents))
-			framework.Logf("Wrote value: \"%v\" to PD %q from pod %q container %q", testFileContents, diskName, host0Pod.Name, containerName)
-
-			// Randomly select a container and read/verify pd contents from it
-			containerName = fmt.Sprintf("mycontainer%v", mathrand.Intn(numContainers)+1)
-			verifyPDContentsViaContainer(f, host0Pod.Name, containerName, fileAndContentToVerify)
-
-			By("deleting host0Pod")
-			framework.ExpectNoError(podClient.Delete(host0Pod.Name, metav1.NewDeleteOptions(0)), "Failed to delete host0Pod")
+	Context("schedule a pod w/ a readonly PD on two hosts, then delete both pods. [Slow]", func() {
+		type testT struct {
+			descr     string                // It description
+			deleteOpt *metav1.DeleteOptions // pod delete option
+		}
+		tests := []testT{
+			{
+				descr:     "immediate (0)",
+				deleteOpt: metav1.NewDeleteOptions(0),
+			},
+			{
+				descr:     "the default (30s)",
+				deleteOpt: &metav1.DeleteOptions{}, // default per provider
+			},
 		}
 
-		By("Test completed successfully, waiting for PD to safely detach")
-		waitForPDDetach(diskName, host0Name)
+		for _, t := range tests {
+			It(fmt.Sprintf("when pod delete grace period is %s", t.descr), func() {
+				framework.SkipUnlessProviderIs("gce", "gke")
```

Comment:
Same as structural nit above.  L174-183 _could_ be in a BeforeEach to separate setup from testing.

Reply:
Agree. This would not reduce the overall lines of code but can improve readability of seeing test "prep" steps separate from the actual test. For this pr I'd prefer not to do this for the reason I mention above.